### PR TITLE
docs + simple hello world example

### PR DESCRIPTION
**Pull request by deian**

### CODE REVIEW

**Comment by deian** [comment]
don't merge at this point. working on docs

File: code/include/rlbox_sandbox.hpp [hunk 1]
```diff
@@ -455,6 +496,17 @@ class rlbox_sandbox : protected T_Sbx
...
   }
 
+  /**
+   * @brief Expose a callback function to the sandboxed code.
+   *
+   * @param func_ptr - The callback to expose.
+   *
+   * @tparam T_RL   - Reference to sandbox.
+   * @tparam T_Ret  - Return type of callback. Must be tainted or void.
+   * @tparam T_Args - Callback argument types. Must be tainted.
```
Comment (shravanrn) [inline code/include/rlbox_sandbox.hpp]: First argument must be rlbox_sandbox<T_Sbx>&, remaining must be tainted.

File: code/include/rlbox_sandbox.hpp [hunk 2]
```diff
@@ -332,16 +364,25 @@ class rlbox_sandbox : protected T_Sbx
...
   {
     return this->impl_is_pointer_in_sandbox_memory(p);
   }
 
+  /**
+   * @brief Check if the pointer points to application memory.
+   */
   inline bool is_pointer_in_app_memory(const void* p)
   {
     return this->impl_is_pointer_in_app_memory(p);
```
Comment (shravanrn) [inline code/include/rlbox_sandbox.hpp]: For the null sandbox both `is_pointer_in_sandbox_memory` and `is_pointer_in_app_memory` return for any pointer as all pointers are both in the sandbox and app memory. For the wasm sandbox, these are mutually exclusive.

File: examples/hello-world-noop/main.cpp [hunk 3]
```diff
@@ -0,0 +1,50 @@
+#define RLBOX_USE_STATIC_CALLS() rlbox_noop_sandbox_lookup_symbol
+
+#include <stdio.h>
+#include "mylib.h"
+#include "../../code/include/rlbox.hpp"
+#include "../../code/include/rlbox_noop_sandbox.hpp"
+
+using namespace rlbox;
+
+void hello_cb(rlbox_sandbox<rlbox_noop_sandbox>& _,
```
Comment (shravanrn) [inline examples/hello-world-noop/main.cpp]: `rlbox_sandbox<rlbox_noop_sandbox>& _` - so haskell :)

**Pull request merged**